src/Components/CreateImageWizard/utilities/useValidation.tsx

```diff
@@ -392,6 +396,49 @@ export function useServicesValidation(): StepValidation {
   };
 }
 
+function validateUserName(userName: string): string {
+  if (!isUserNameValid(userName)) {
+    return 'Invalid user name';
+  }
+  return '';
+}
+
+function validatePassword(
+  userPassword: string,
+  environments: string[]
+): string {
+  const DEFAULT_PASSWORD_ERROR_MESSAGE =
+    'Password helps protect your account, we recommend a password of at least 6 characters.';
+  const AZURE_PASSWORD_ERROR_MESSAGE =
+    "A password for the target environment 'Azure' must be at least 6 characters long. Please enter a longer password.";
+  const WEAK_PASSWORD_WARNING =
+    'WARNING: This password seems weak, please use with caution or include at least 3 of the following: lowercase letter, uppercase letters, numbers, symbols';
+
+  const passwordValidationResult = isPasswordValid(userPassword);
+  const isAzureEnvironment = environments.includes('azure');
+
+  if (isAzureEnvironment && !passwordValidationResult.isValid) {
+    return AZURE_PASSWORD_ERROR_MESSAGE;
+  }
+
+  if (isAzureEnvironment && passwordValidationResult.strength === 'error') {
+    return WEAK_PASSWORD_WARNING;
+  }
+
+  if (!passwordValidationResult.isValid) {
+    return DEFAULT_PASSWORD_ERROR_MESSAGE;
+  }
+
+  return '';
+}
+
+function validateSshKey(userSshKey: string): string {
```

Comment:
This function doesn't validate the SSH key, it just returns the error message for an invalid key. Naming is very hard, but a good starting point is to name functions after what they do.